### PR TITLE
NoUnusedCapturingCatchFixer - introduction

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: src/Fixer/ControlStructure/NoUnusedCapturingCatchFixer.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    /**
+     * @param int $index
+     * @param int $startIndex
+     * @param int $endIndex
+     *
+     * @return bool
+     */
+    private function isCapturePossiblyUsed(Tokens $tokens, $index, $startIndex, $endIndex)
+    {
+        static $riskyKinds = [
```
Comment (GrahamCampbell) [inline src/Fixer/ControlStructure/NoUnusedCapturingCatchFixer.php:106]: Duplicated logic with the unused closure use fixer?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Fixer/ControlStructure/NoUnusedCapturingCatchFixer.php (hunk 1 above)
Comment (SpacePossum) [reply src/Fixer/ControlStructure/NoUnusedCapturingCatchFixer.php]: :+1:  good point, extracted, continuing the work now

**Comment by GrahamCampbell** [comment]
One has to be careful the variable isn't already defined before the catch block, otherwise not overriting it is wrong. In particular, this fixer can only ever be applied in function or method bodies, and not just in global land.

**Comment by GrahamCampbell** [comment]
Imagine

```php
$e = 123;

try {
    throw new Exception();
} catch (Exception $e)
{}

var_dump($e);
```

is not the same as:

```php
$e = 123;

try {
    throw new Exception();
} catch (Exception)
{}

var_dump($e);
```

**Comment by SpacePossum** [comment]
Yeah, I'm not sure yet to either mark this fixer as risky or go about to extend the scope of checking if the variable is used after the catch block as well.

**Comment by keradus** [comment]
I would mark it as risky and have it done.

**Pull request closed without merging**